metal/v1/model_vlan_virtual_circuit.go

```diff
@@ -433,6 +433,46 @@ func (o *VlanVirtualCircuit) SetUpdatedAt(v time.Time) {
 	o.UpdatedAt = &v
 }
 
+// assertRequired throws an error if a required field is nil
+// It does not check nested models
+func (o *VlanVirtualCircuit) assertRequired() error {
+	if IsNil(o.Bill) {
+		return MissingRequiredFieldError("bill")
+	}
+	if IsNil(o.Description) {
+		return MissingRequiredFieldError("description")
+	}
+	if IsNil(o.Id) {
+		return MissingRequiredFieldError("id")
+	}
+	if IsNil(o.Name) {
+		return MissingRequiredFieldError("name")
+	}
+	if IsNil(o.NniVlan) {
+		return MissingRequiredFieldError("nni_vlan")
+	}
+	if IsNil(o.Port) {
+		return MissingRequiredFieldError("port")
+	}
+	if IsNil(o.Project) {
+		return MissingRequiredFieldError("project")
+	}
+	if IsNil(o.Status) {
+		return MissingRequiredFieldError("status")
+	}
+	if IsNil(o.Tags) {
+		return MissingRequiredFieldError("tags")
+	}
+	if IsNil(o.VirtualNetwork) {
+		return MissingRequiredFieldError("virtual_network")
+	}
+	if IsNil(o.Vnid) {
```

Comment:
A potential complication with required field validation (doesn't necessarily apply here in a business logic sense, but  it does in a type sense):

`o.Vnid` is a required integer field, so it's generated as an `int32` (as opposed to `*int32` for optional fields); this means that we can't tell if `o.Vnid` is `0` because the actual value is `0` or because it is unset.  In this case I think that's not technically an issue, but if `0` were a valid value for `o.Vnid`, then this change would throw a validation error on a valid value.

Reply:
Path schema parameters can define `allowEmpty`. It doesn't help us here, but I wonder if these Assertion tests would run up against parameter types and need to consider this property.

Reply:
Code related to `required` parameters is generated elsewhere. The model template modified here only applies to `required` as it appears in schema objects.